### DISCUSSION TITLE
How to limit joint velocities in task velocity

### DISCUSSION THREAD

**Discussion by NunoDuarte** [robotology/community#249] [Q&A] [answered]
Hi everyone,

I'm using Cartesian interface to control in task velocities using the following function:

`icart->setTaskVelocities(xdot,odot)`

I was wondering if the joint velocities are limited or not and if I can set a limit so I won't break the iCub :)

Thanks in advance,
Nuno

**Comment by pattacini** [comment, marked as answer]
Hi @NunoDuarte 

#208 does pretty much explain how `setTaskVelocities()` works.

To limit the reactivity of the controller, therefore, do play with the `Ttraj` parameter by calling the method [`setTrajTime()`][1]. Higher `Ttraj` corresponds to slower joint velocities.

[1]: http://www.yarp.it/classyarp_1_1dev_1_1ICartesianControl.html#a46ede2ac6bed5e0e459deb843a8f7783

**Comment by NunoDuarte** [comment]
Hi Pattacini,

Sorry for the late response. I saw your comment. I thought there was another way to control the joint velocities (like setRefVelocities) but I think that will do. :)

Thanks

**Comment by pattacini** [comment]
No worries @NunoDuarte 

The idea is that you don't see the mechanism (i.e. joints) underneath, but rather you keep dealing with the end-effector only 😉 

How the joints move in terms of direction and speed is a detail that should remain hidden in the implementation.
